lib/ansible/modules/notification/mattermost.py

```diff
@@ -0,0 +1,160 @@
+    required: false
+    default: "Ansible"
+  icon_url:
+    description:
+      - Url for the message sender's icon (default C(https://www.ansible.com/favicon.ico))
+    required: false
+  validate_certs:
+    description:
+      - If C(no), SSL certificates will not be validated. This should only be used
+        on personally controlled sites using self-signed certificates.
+    required: false
+    default: 'yes'
+    choices:
+      - 'yes'
+      - 'no'
+"""
+
+EXAMPLES = """
+- name: Send notification message via Mattermost
+  mattermost:
+    url: http://mattermost.example.com
+    api_key: my_api_key
+    text: '{{ inventory_hostname }} completed'
+
+- name: Send notification message via Slack all options
+  mattermost:
+    url: http://mattermost.example.com
+    api_key: my_api_key
+    text: '{{ inventory_hostname }} completed'
+    channel: notifications
+    username: 'Ansible on {{ inventory_hostname }}'
+    icon_url: http://www.example.com/some-image-file.png
+"""
+
+RETURN = '''
+payload:
+    description: Mattermost payload
+    returned: success
+    type: string
+webhook_url:
+    description: URL the webhook is sent to
+    returned: success
+    type: string
+'''
+
+
+def main():
+    module = AnsibleModule(
+        supports_check_mode=True,
+        argument_spec = dict(
+            url         = dict(type='str', required=True),
+            api_key     = dict(type='str', required=True,  no_log=True),
+            text        = dict(type='str', required=True),
+            channel     = dict(type='str', required=False,  default=None),
+            username    = dict(type='str', default='Ansible'),
+            icon_url    = dict(type='str', default='https://www.ansible.com/favicon.ico'),
+            validate_certs = dict(default='yes', type='bool'),
+        )
+    )
+    #init return dict
+    retkwargs = dict(changed=False, msg="OK")
+
+    #define webhook
+    webhook_url = "{0}/hooks/{1}".format(module.params['url'],module.params['api_key'])
+    retkwargs['webhook_url'] = webhook_url
+
+    #define payload
+    payload = { }
+    for param in ['text', 'channel', 'username', 'icon_url']:
+        if module.params[param] is not None:
+            payload[param] = module.params[param]
+
+    payload=module.jsonify(payload)
+    retkwargs['payload'] = payload
+
+    #http headers
+    headers = {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+    }
+
+    #send request
+    response, info = fetch_url(module=module, url=webhook_url, headers=headers, method='POST', data=payload)
+
+
+    if info['status'] != 200:
+        #some problem
+        retkwargs['msg'] = "Failed to send mattermost message, the error was: {0}".format(info['msg'])
+        module.fail_json(**retkwargs)
+
+    #Looks good
+    module.exit_json(**retkwargs)
+
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
```

Comment:
Nowadays, we list the imported functionality at the top of the module, and we explicitly list all the functionality we import. So in your case it would likely become:
```python
from ansible.module_utils.basic import AnsibleModule
from ansible.module_utils.urls import fetch_url
```